geomesa-tools/src/test/scala/org/locationtech/geomesa/tools/DataStoreRegistration.scala

```diff
@@ -0,0 +1,81 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.tools
+
+import java.awt.RenderingHints
+import java.util.Collections
+import java.util.concurrent.ConcurrentHashMap
+
+import org.geotools.data.DataAccessFactory.Param
+import org.geotools.data.{DataStore, DataStoreFactorySpi}
+import org.geotools.factory.FactoryIteratorProvider
+
+/**
+  * For registering a data store instance that can be loaded via SPI
+  */
+object DataStoreRegistration {
```

Comment:
Ok, I'm reading quickly and am stumped.  Why is this useful?

Reply:
I added a better comment - it's a test class, used to pass a MemoryDataStore instance through SPI loading. Otherwise the data store is loaded, but any features written are lost b/c they're only kept in memory, so you can't verify anything actually happened.